linux/src/threadpool_linux.c

```diff
@@ -379,7 +377,7 @@ THANDLE(THREADPOOL) threadpool_create(EXECUTION_ENGINE_HANDLE execution_engine)
                             }
                             else
                             {
-                                (void)interlocked_exchange(&result->state, 0);
+                                (void)interlocked_exchange(&result->state, THREADPOOL_STATE_STARTED);
```

Comment:
>sult->state, THREADPOOL_STATE_STARTED); [](http://example.com/codeflow?start=62&length=39)

Honestly I think that this state can only map to a state machine.
Now, the state machine can have states that align to the SM states or not.

The way I read the code it seems that this state machine now is a sub-state machine which maps the OPEN state of SM.

In other words, when we are in open, we can be in one of the 2 sub-states: stopped or started.
If that is the case the state should always be set to STARTED when we open with sm. The sub-state means nothing outside of sm OPEN.

If somehow this state (and state machine applies to other sm states, we kind of have a problem, because then we'd be having 2 state machines running in parallel.

So, I suspect this assignment is wrong (or something deeper exists as an issue here).
Shall we talk?

Reply:
Not really a states machine, it's more of a starting and stopping of the thread.  I'll revert this one and then we can talk about it.